File: src/main/java/tk/skulk/plugin/core/extensions/silencemobs/listeners/PlayerInteractEntityListener.java
package tk.skulk.plugin.core.extensions.silencemobs.listeners;

import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
import org.bukkit.GameMode;
import org.bukkit.Material;
import org.bukkit.entity.EntityType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.player.PlayerInteractEntityEvent;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.extensions.silencemobs.SilenceMobsExtension;
import tk.skulk.plugin.wrapper.BaseListener;

import static tk.skulk.plugin.helpers.MiniMessageHelper.fmt;
import static tk.skulk.plugin.helpers.MiniMessageHelper.sendMessage;

public final class PlayerInteractEntityListener extends BaseListener<SilenceMobsExtension> {
    public PlayerInteractEntityListener(final @NotNull SilenceMobsExtension extension) {
        super(extension);
    }

    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    public void onPlayerInteractEntity(final @NotNull PlayerInteractEntityEvent event) {
        final var player = event.getPlayer();
        final var playerInventory = player.getInventory();
        final var mainHandItem = playerInventory.getItemInMainHand();
        final var offHandItem = playerInventory.getItemInOffHand();

        final ItemStack nameTags;

        if (mainHandItem.getType() == Material.NAME_TAG) {
            nameTags = mainHandItem;
        }
        else if (offHandItem.getType() == Material.NAME_TAG) {
            nameTags = offHandItem;
        }
        else {
            return;
        }

        if (!PlainTextComponentSerializer.plainText().serialize(nameTags.displayName()).equalsIgnoreCase(
            "[silenced]")) {
            return;
        }

        final var entity = event.getRightClicked();

        if (entity.getType() == EntityType.UNKNOWN) {
            return;
        }

        entity.setSilent(true);
        entity.customName(fmt("<i><gray>Silenced</gray></i>"));

        if (player.getGameMode() != GameMode.CREATIVE) {
            nameTags.setAmount(nameTags.getAmount() - 1);
        }

        sendMessage(
            player,
            "green",
            '✓',
            "The entity '<0>' has been silenced.",
            entity.getType().toString().toLowerCase()
        );

        event.setCancelled(true);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/silenceMobs/listeners/PlayerInteractEntityListener.kt
package tk.skulk.plugin.core.extensions.silenceMobs.listeners

import org.bukkit.GameMode
import org.bukkit.Material
import org.bukkit.entity.EntityType
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.player.PlayerInteractEntityEvent
import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.silenceMobs.SilenceMobsExtension
import tk.skulk.plugin.framework.Listener
import tk.skulk.plugin.util.asString
import tk.skulk.plugin.util.format
import tk.skulk.plugin.util.sendMessage

class PlayerInteractEntityListener(extension: SilenceMobsExtension) : Listener<SNPlugin, SilenceMobsExtension>(
    extension
) {
    @EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
    fun onPlayerInteractEntity(event: PlayerInteractEntityEvent) {
        val player = event.player

        val nameTags = when {
            player.inventory.itemInMainHand.type == Material.NAME_TAG -> player.inventory.itemInMainHand
            player.inventory.itemInOffHand.type == Material.NAME_TAG -> player.inventory.itemInOffHand
            else -> return
        }

        if (nameTags.displayName().asString().equals("[Silenced]", ignoreCase = true)) {
            return
        }

        val entity = event.rightClicked

        if (entity.type == EntityType.UNKNOWN) {
            return
        }

        entity.isSilent = true
        entity.customName(format("<i><gray>Silenced</gray></i>"))

        if (player.gameMode != GameMode.CREATIVE) {
            nameTags.amount -= 1
        }

        player.sendMessage(
            "green", '✓', "The entity '<0>' has been silenced.", entity.type.toString().lowercase()
        )

        event.isCancelled = true
    }
}
